app/redux/slices/settingsSlice.ts

```diff
@@ -1,21 +1,31 @@
 import { createSlice } from '@reduxjs/toolkit';
-
-import { ISettingsActionPayload } from '../types';
+import {
+  ILanguageActionPayload,
+  INotificationAllowActionPayload,
+} from '../types';
 
 interface ISettingsState {
   language: string;
+  isNotificationAllow: boolean;
```

Comment:
rename to `isNotificationsEnabled`
all settings options should be with suffix `Enabled` for switches